spark/common/src/test/scala/org/apache/sedona/sql/rasteralgebraTest.scala

```diff
@@ -1296,6 +1296,58 @@ class rasteralgebraTest extends TestBaseScala with BeforeAndAfter with GivenWhen
       assertEquals("UNSIGNED_8BITS", result)
     }
 
+    it("Passed RS_MapAlgebra with two raster columns") {
+      val df1 = sparkSession.read.format("binaryFile")
+        .option("recursiveFileLookup", "true")
+        .option("pathGlobFilter", "*.tif*")
+        .load(resourceFolder + "raster")
+        .selectExpr("path", "RS_FromGeoTiff(content) as rast0")
+      val df2 = sparkSession.read.format("binaryFile")
```

Comment:
No need to read the same dataset twice. Please use self-join: https://www.sparkcodehub.com/spark-self-join-in-spark-sql-and-dataframe